lib/pact/consumer/mock_service/candidate_options.rb

```diff
@@ -0,0 +1,22 @@
+require 'pact/consumer/mock_service/web_request_options'
+
+module Pact
+  module Consumer
+
+    # Allow web preflight requests to the intaractions setup by the user
+    # This is only needed in a CORS setup, where the browsers do
+    # an OPTIONS call before a DELETE, POST (for most request), etc. in a cross domain requests
+    class CandidateOptions < WebRequestOptions
+      def initialize name, logger, cors_enabled
+        super(name,logger)
+        @cors_enabled = cors_enabled
+      end
+
+      # Will match all requests to OPTIONS when in CORS mode
+      def request_path_match? env
+        @cors_enabled
```

Comment:
The following is defined in the parent
def request_method
'OPTIONS'
end

With its own parent having the matcher defined

def match? env
(request_header_match? env) && (request_path_match? env) && (request_method_match? env)
end